iceoryx2-pal/posix/Cargo.toml

```diff
@@ -19,11 +19,7 @@ acl = []
 
 [dependencies]
 iceoryx2-pal-concurrency-sync = { workspace = true }
-iceoryx2-pal-configuration = { workspace = true }
```

Comment:
Those dependencies are required by mac os and windows.

Reply:
Do you have time to add the `cargo udeps` check also to our CI target in `.github/workflows/build-test.yml` in the preflight check.
You do not have to do this in this PR but it would be nice when the effort you put in here would persist and permanently improve our dependency quality.

Anyway, thank you for your contribution!

Reply:
It seems building `cargo-udeps´ takes quite long. If you do this, please have a look at the cashing of `cargo-nextest` and dot it similarly.

Reply:
I'll take a look what could be done.